extensions/azcli/src/az.ts

```diff
@@ -285,36 +267,70 @@ async function executeAzCommand(command: string, args: string[], additionalEnvVa
 	return executeCommand(command, args, additionalEnvVars);
 }
 
-// Commented the install/update prompts out until DoNotAskAgain is implemented
-// async function setConfig(key: string, value: string): Promise<void> {
-// 	const config = vscode.workspace.getConfiguration(azConfigSection);
-// 	await config.update(key, value, vscode.ConfigurationTarget.Global);
-// }
+function getConfig(key: string): AzDeployOption | undefined {
+	const config = vscode.workspace.getConfiguration(azConfigSection);
+	const value = <AzDeployOption>config.get<AzDeployOption>(key);
+	return value;
+}
+
+async function setConfig(key: string, value: string): Promise<void> {
+	const config = vscode.workspace.getConfiguration(azConfigSection);
+	await config.update(key, value, vscode.ConfigurationTarget.Global);
+}
+
 
 /**
- * Find user's local Azure CLI. Execute az --version and parse out the version number.
- * If an update is needed, prompt the user to update via link. Return the AzTool.
- * Currently commented out because Don't Prompt Again is not properly implemented.
+ * Finds and returns the user's locally installed Azure CLI tool. Checks to see if arcdata extension is
+ * installed and updated. If no Azure CLI is found, the user will be prompted to install it and an
+ * undefined object is returned.
  */
-async function findSpecificAz(): Promise<IAzTool> {
+export async function findAzAndCheckArcdata(): Promise<IAzTool | undefined> {
 	const path = await ((process.platform === 'win32') ? searchForCmd('az.cmd') : searchForCmd('az'));
+
+	let response: string | undefined;
+	const responses = [loc.askLater, loc.doNotAskAgain];
+
+	if (path === undefined) {
+		// If no Azure CLI was found, prompt to install
+		const azInstallNeededConfig = <AzDeployOption>getConfig(azCliInstallKey);
+		if (azInstallNeededConfig !== AzDeployOption.dontPrompt) {
+			response = await vscode.window.showErrorMessage(loc.noAzureCLI, ...responses);
```

Comment:
So what did you do to fix the test issue? It still looks like you're blocking activation here on user input from what I can tell so I'm a little confused how it's passing now. Or was I mistaken on the source of the issue?

Reply:
Basically my findAz() function checks to see if user has Azure CLI installed, and if not then prompt them to follow the link to download or Don't Ask Again, and then checks arcdata extension for install and update as well with prompts. I added a param to findAz() called userRequested which mimics the behavior in the azdata extension; it's set to false by default, and if false then the prompts will be skipped. But because findAz() is only really called in activate(), this essentially disables the prompts for the time being since it will always be false. For post-GA plans I'm going to be adding back the automatic updates/installs so it will not always be the case that userRequested is false after that.

Reply:
So you were right about the test failing due to extension activate waiting for user input (Don't Ask Again option), I just resolved this issue by temporarily disabling the code that prompts for that. But in a future PR the prompts will be re-enabled as I change how these findAz() update/install functions look. This is part of the original plan to eventually merge azcli back into arc to make the user experience of installing the arc extension less cumbersome than it is right now

Reply:
Hmm, I guess that works. But the preferred way would be not to have activate block on finding the az installation. That's what we do in azdata currently 

https://github.com/microsoft/azuredatastudio/blob/main/extensions/azdata/src/extension.ts#L38

(it doesn't await the Promise)

Extension activation time greatly impacts the user experience - and so having it block on potentially long running operations (and especially potentially infinite ones like waiting for user input) is strongly recommended not to do. Instead the extension should kick off stuff async'ly that it needs to and then make sure the code handles that stuff not being completed yet when it tries to access that functionality later on.

Reply:
I would like to do it async-ly but when I remove the await from the findAz() I get a build error that says: 
Type 'Promise<IAzTool | undefined>' is missing the following properties from type 'IAzTool': executeCommand, arcdata, postgres, sql, and 4 more

Can this function be called async-ly?

Reply:
I assume you're trying to do this?

`	azToolService.localAz = findAz();`

? Because yes, that won't work since findAz returns a promise. You need to do what azdata does and set localAz inside of a .then 

```
findAz().then(async azTool => {
    if (azTool !== undefined) {
	azToolService.localAz = azTool ;
...
```

Reply:
Thought I should leave a comment here, I think even though these tests pass, I am going to abandon this PR so that I can instead add back Don't Prompt Again when I re-enable the automatic updating and installing to mimic the original azdata extension. My earlier commit adding the userRequested bool would deactivate all Don't Prompt Again toasts anyways, so I think it would be better to send a PR when everything is ready. If you think this is a good plan going forward, let me know and I will continue with the PR abandon.

Reply:
Sounds good. If you wanted to get the prompt back in some form quickly let me know and I can help with that